city_scrapers/spiders/losca_City_Planning.py

```diff
@@ -0,0 +1,101 @@
+import re
+
+from city_scrapers_core.constants import BOARD, COMMISSION, NOT_CLASSIFIED
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from dateutil.parser import parse
+
+
+class LoscaCityPlanningSpider(CityScrapersSpider):
+    name = "losca_City_Planning"
+    agency = "Los Angeles City Planning"
+    timezone = "America/Los_Angeles"
+    start_year = 2024
+    start_urls = [
+        f"https://planning.lacity.gov/dcpapi/meetings/api/all/commissions/{start_year}",
+        f"https://planning.lacity.gov/dcpapi/meetings/api/all/boards/{start_year}",
+        f"https://planning.lacity.gov/dcpapi/meetings/api/all/hearings/{start_year}",
+    ]
+
+    def parse(self, response):
+        """
+        This spider retrieves meetings for the specified `start_year`.
+        Update `start_year` to change the target year.
+        """
+        try:
+            data = response.json()
+            if "Entries" not in data:
+                self.logger.error("Missing 'Entries' field in response")
+                return
+            items = data["Entries"]
+        except ValueError as e:
+            self.logger.error(f"Failed to parse JSON response: {e}")
+            return
+
+        for item in items:
+            if not all(key in item for key in ["Type", "Date", "Address"]):
+                self.logger.warning(
+                    f"Skipping item due to missing required fields: {item}"
+                )
+                continue
+            meeting = Meeting(
+                title=item.get("Type", ""),
+                description=item.get("Note", ""),
+                classification=self._parse_classification(item),
+                start=parse(item.get("Date")),
+                end=None,
+                all_day=False,
+                time_notes="",
+                location=self._parse_location(item),
+                links=self._parse_links(item),
+                source=self._parse_source(response),
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
+        pattern = r"(\d{3,4}(?:\s?–\s?\d{3,4})?(?:\s?and\s?\d{3,4})?(?:\s?,?\s?\d{3,4})?\s[\w\s]+\b(?:Road|Drive|Boulevard|Avenue))"  # noqa
+        addresses = re.findall(pattern, item["Address"])
+        address = ", ".join(addresses) if addresses else item.get("Address", "")
+        return {
+            "address": address,
+            "name": item.get("BoardName", ""),
+        }
+
+    def _validate_url(self, url):
+        """Validate URL format and security."""
+        if not url:
+            return False
+        if not url.startswith("https://"):
+            self.logger.warning(f"Non-HTTPS URL found: {url}")
```

Comment:
All the links returned from the API are "http". But when you visit them, they will be redirected to "https" protocol. So your validation here for https URLs is too strict, which results in no links are scraped. Please lessen the validation to accept both "http" and "https" URLs.